diff --git a/ext/native/ui/root.cpp b/ext/native/ui/root.cpp
index 86b121f9c..b961fac21 100644
--- a/ext/native/ui/root.cpp
+++ b/ext/native/ui/root.cpp
@@ -295,9 +295,9 @@ bool AxisEvent(const AxisInput &axis, ViewGroup *root) {
 		}
 		if (axis.axisId == JOYSTICK_AXIS_Y) {
 			if (axis.value < -THRESHOLD)
-				dir = DIR_NEG;
-			else if (axis.value > THRESHOLD)
 				dir = DIR_POS;
+			else if (axis.value > THRESHOLD)
+				dir = DIR_NEG;
 			// We stupidly interpret the joystick Y axis backwards on Android instead of reversing
 			// it early (see keymaps...). Too late to fix without invalidating a lot of config files, so we
 			// reverse it here too.
